src/Workspaces/Core/Portable/Workspace/Solution/AssetHint.cs

```diff
@@ -13,19 +13,42 @@ namespace Microsoft.CodeAnalysis.Serialization;
 [DataContract]
 internal readonly struct AssetHint
 {
-    public static readonly AssetHint None = default;
+    /// <summary>
+    /// Instance that will only look up solution-level data when searching for checksums.
+    /// </summary>
+    public static AssetHint SolutionOnly = default;
+
+    /// <summary>
+    /// Instance that will only look up solution-level, as well as the top level nodes for projects when searching for
+    /// checksums.  It will not descend into projects.
+    /// </summary>
+    public static AssetHint SolutionAndTopLevelProjectsOnly = new(projectId: null, documentId: null, topLevelProjects: true, isFullLookup_ForTestingPurposesOnly: false);
+
+    /// <summary>
+    /// Special instance, allowed only in tests/debug-asserts, that can do a full lookup across the entire checksum
+    /// tree.  Should not be used in normal release-mode product code.
+    /// </summary>
+    public static AssetHint FullLookupForTesting = new(projectId: null, documentId: null, topLevelProjects: false, isFullLookup_ForTestingPurposesOnly: true);
 
     [DataMember(Order = 0)]
     public readonly ProjectId? ProjectId;
     [DataMember(Order = 1)]
     public readonly DocumentId? DocumentId;
+    [DataMember(Order = 3)]
+    public readonly bool TopLevelProjects;
+    [DataMember(Order = 4)]
+    public readonly bool IsFullLookup_ForTestingPurposesOnly;
```

Comment:
>IsFullLookup_ForTestingPurposesOnly; [](http://example.com/codeflow?start=25&length=36)

This doesn't feel right. Might feel cleaner if this and the other bool were an enum or flag.

Reply:
let me try tthat.

Reply:
thanks!  that was cleaner.